### DISCUSSION TITLE
the option to add collection inside another One

### DISCUSSION THREAD

**Discussion by regisx001** [pocketbase/pocketbase#386] [Q&A]
it will be great to add the option to create a collection inside a record , it will manage the structure of the project , it like a folder inside a Record , it gonna be great

**Comment by ganigeorgiev** [comment]
Hm, I guess this could be used for creating small inline collection(s) that you don't want to be visible in the sidebar, but I'm not sure how useful that would be.

Could you give an example how do you plan to use it (pseudo code is also welcomed)?

**Comment by regisx001** [comment]
I have been used firebase and I saw this option in the Cloudfire store,
Like it shown in the picture below :
![Screenshot from 2022-09-05 07-09-43](https://user-images.githubusercontent.com/98271602/188372197-7fa106aa-27ed-4e30-8c9b-ea6e102d0b9e.png)

i tell you a simple example :
i want to make a todo application (the users will store the same data schema in the database)
I will make a collection named Users, and make a user relation for the first collection Row .
and after that we add another collection inside the 1st one name todos (that have the another schema),
in conclusion : when the user create a account , it make a record in the first Collection and when he submit the data it stored in second one .
That How it will be !
I did not find a solution for that in the actuel collection option !
and i will use it to create a Movie app for the desktop env  using flutter it will be useful

**Comment by ganigeorgiev** [comment]
In PocketBase you can do almost the same if you use the `relation` type field (you have to create the related collection beforehand so that you can link it).

The only drawback of the current approach is that you can't submit the 2 collection records in a single transaction, or in other words, if you have 2 related collections you have to submit each of them separately, eg.:

```js
const project = await client.records.create("projects", {
    ...
});

const todo = await client.records.create("todos", {
    project: project.id,
    ...
});
```

There are plans to add support for transactional create/update in https://github.com/pocketbase/pocketbase/issues/48, but that's currently on hold. The idea is to allow submitting/importing records from different collections in a single transactional request (aka. if one fail, nothing get saved), something like:

```js
const result = await client.records.bulkCreate({
    projects: [ {...}, {...} ],
    todos: [ {...}, {...} ],
});
```

---

_I'll move the issue to Q&A discussion._
